astropy/io/ascii/ecsv.py

```diff
@@ -54,14 +54,6 @@ def write(self, lines):
         if self.splitter.delimiter not in DELIMITERS:
             raise ValueError('only space and comma are allowed for delimiter in ECSV format')
 
-        for col in self.cols:
-            if len(getattr(col, 'shape', ())) > 1:
-                raise ValueError(
-                    f"ECSV format does not support multidimensional column '{col.info.name}'\n"
-                    'One can filter out such columns using:\n'
-                    'names = [name for name in tbl.colnames if len(tbl[name].shape) <= 1]\n'
-                    'tbl[names].write(...)')
-
```

Comment:
Does this mean that such tables will be picked up by the ECSV writer so that calling `ascii
write` will use the ECSV format by default?

Reply:
Good point. This won't automatically use ECSV by default, so in that case (`ascii.write(t)`) the issue #11148 of useless error messages will still occur. So I'll take the opportunity to fix that in this PR with a message telling the user to use ECSV.

Reply:
Great, that's what I was concerned about.